backend/scripts/rerun_failing_test_executions.py

```diff
@@ -0,0 +1,87 @@
+# ruff: noqa: T201
+
+import argparse
+import re
+
+import requests
+
+TO_API_URL = "https://test-observer-api.canonical.com"
+
+
+def main(artefact_id: int, test_case_regex: str, reversed: bool):
+    artefact_builds = requests.get(
+        f"{TO_API_URL}/v1/artefacts/{artefact_id}/builds"
+    ).json()
+
+    test_case_matcher = re.compile(test_case_regex)
+
+    relevant_test_executions = [
+        te
+        for ab in artefact_builds
+        for te in ab["test_executions"]
+        if te["review_decision"] == []
+        and not te["is_rerun_requested"]
+        and te["status"] == "FAILED"
+    ]
+
+    test_execution_ids_to_rerun = []
+    for te in relevant_test_executions:
+        test_results = requests.get(
+            f"{TO_API_URL}/v1/test-executions/{te['id']}/test-results"
+        ).json()
+
+        matching_failed_tests = (
+            tr
+            for tr in test_results
+            if tr["status"] == "FAILED" and test_case_matcher.match(tr["name"])
```

Comment:
Probably you can introduce a small pydantic model at the top of the script instead of dealing with raw json

Reply:
This is meant to be a small script that we should get rid of later on (as everything should be through the api or frontend). I don't think it's necessary to have this added complexity